Filtered_kernel/include/CGAL/Converting_construction.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022  GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Laurent Rineau
+
+#ifndef CGAL_CONVERTING_CONSTRUCTION_H
+#define CGAL_CONVERTING_CONSTRUCTION_H
+
+#include <CGAL/config.h>
+#include <CGAL/Cartesian_converter.h>
+#include <CGAL/Cartesian/Is_trivial_construction.h>
+#include <type_traits>
+
+namespace CGAL {
+
+template <class Source_construction, class Target_construction,
+          class Converter, class Backward_converter>
+struct Converting_construction
+{
+  CGAL_NO_UNIQUE_ADDRESS Source_construction source_construction;
+  CGAL_NO_UNIQUE_ADDRESS Target_construction construct;
+  CGAL_NO_UNIQUE_ADDRESS Converter convert;
+  CGAL_NO_UNIQUE_ADDRESS Backward_converter backward_convert;
+
+  template <typename... Args,
+            std::enable_if_t<CartesianFunctors::Is_trivial_construction<Source_construction, Args...>::value>* = nullptr>
+  decltype(auto) operator()(Args&&... args) const {
+    return source_construction(std::forward<Args>(args)...);
+  }
+
+  template <typename... Args,
+            std::enable_if_t<!CartesianFunctors::Is_trivial_construction<Source_construction, Args...>::value>* = nullptr>
+  decltype(auto) operator()(Args&&... args) const {
+    return backward_convert(construct(convert(args)...));
```

Comment:
This line generates a warning with visual studio `C4172 : returning address of local variable or temporary`